wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.java

```diff
@@ -272,7 +272,24 @@ public String toString(E value) {
       return reader.readVarint64();
     }
   };
-  public static final ProtoAdapter<Long> UINT64 = INT64;
+  /**
+   * Like INT64, but negative longs are interpreted as large positive values, and encoded that way
+   * in JSON.
+   */
+  public static final ProtoAdapter<Long> UINT64 = new ProtoAdapter<Long>(
```

Comment:
... And here it matters! Because we're using the type adapter to imply what the JSON encoding should be. The binary encoding for both of these types is the same, but the JSON encoding is different.